### DISCUSSION TITLE
6.15.0 breaks dump-autoload

### DISCUSSION THREAD

**Discussion by html5maker** [spatie/laravel-backup#1258] [Q&A]
After installing 6.15.0 version dump-autoload fails with the following error:

```
In Container.php line 805:
                                       
  Target class [hash] does not exist.  
                                       
In Container.php line 803:
                             
  Class hash does not exist  
```

It's probably because it requires a specific ext-zip version ("ext-zip": "^1.14.0") that is incompatible with other libraries...

After rollback to 6.14.4 everything works fine...

**Comment by vbarbarosh** [comment]
This is an issue with `gcr.io/google-appengine/php73:latest`. It uses obsolete zip library.

Here is how to reproduce:

    $ docker run --rm -i gcr.io/google-appengine/php73:latest bash << EOF
        cd
        composer self-update --2
        composer create-project laravel/laravel
        cd laravel
        composer require spatie/laravel-backup:6.15
    EOF

    [...]

    Generating optimized autoload files
    > Illuminate\Foundation\ComposerScripts::postAutoloadDump
    > @php artisan package:discover --ansi

       Error 

      Undefined class constant 'EM_AES_256'

      at vendor/spatie/laravel-backup/config/backup.php:138
        134▕         /*
        135▕          * The encryption algorithm to be used for archive encryption.
        136▕          * You can set it to `null` or `false` to disable encryption.
        137▕          */
      ➜ 138▕         'encryption' => \ZipArchive::EM_AES_256,
        139▕     ],
        140▕ 
        141▕     /*
        142▕      * You can get notified when specific events occur. Out of the box you can use 'mail' and 'slack'.

          +9 vendor frames 
      10  artisan:37
          Illuminate\Foundation\Console\Kernel::handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
    Script @php artisan package:discover --ansi handling the post-autoload-dump event returned with error code 1

And a version of zip it was packaged with:

    $ docker run --rm -i gcr.io/google-appengine/php73:latest bash <<< 'php -i | grep zip'
    Configure Command =>  './configure'  '--build=x86_64-linux-gnu' '--prefix=/usr' '--includedir=${prefix}/include' '--mandir=${prefix}/share/man' '--infodir=${prefix}/share/info' '--sysconfdir=/etc' '--localstatedir=/var' '--disable-silent-rules' '--libdir=${prefix}/lib/x86_64-linux-gnu' '--libexecdir=${prefix}/lib/x86_64-linux-gnu' '--disable-maintainer-mode' '--disable-dependency-tracking' '--prefix=/opt/php73' '--with-config-file-path=/opt/php73/lib' '--with-config-file-scan-dir=/opt/php73/lib/ext.enabled:/opt/php73/lib/conf.d' '--enable-sysvsem' '--enable-sysvshm' '--enable-sysvmsg' '--disable-cgi' '--enable-bcmath=shared' '--enable-calendar=shared' '--enable-exif=shared' '--enable-fpm' '--enable-ftp=shared' '--enable-intl=shared' '--enable-mbstring' '--enable-mysqlnd' '--enable-opcache-file' '--enable-pcntl' '--enable-shared' '--enable-shmop=shared' '--enable-soap=shared' '--enable-sockets' '--enable-zip' '--with-bz2' '--with-curl' '--with-gettext=shared' '--with-gd=shared' '--with-gmp' '--with-freetype-dir=/usr' '--with-jpeg-dir=/usr' '--with-mcrypt' '--with-pdo_sqlite=shared,/usr' '--with-pdo-pgsql' '--with-pgsql' '--with-sqlite3=shared,/usr' '--with-xmlrpc=shared' '--with-xsl=shared' '--with-fpm-user=www-data' '--with-fpm-group=www-data' '--with-mysql' '--with-mysqli=mysqlnd' '--with-pdo-mysql=mysqlnd' '--with-openssl' '--with-pcre-regex' '--with-readline' '--with-recode' '--with-zlib' 'build_alias=x86_64-linux-gnu'
    Registered PHP Streams => https, ftps, compress.zlib, compress.bzip2, php, file, glob, data, http, ftp, zip, phar
    Registered Stream Filters => zlib.*, bzip2.*, convert.iconv.*, string.rot13, string.toupper, string.tolower, string.strip_tags, convert.*, consumed, dechunk
    Stream Wrapper support => compress.bzip2://
    Stream Filter support => bzip2.decompress, bzip2.compress
    gzip compression => enabled
    bzip2 compression => enabled
    zip
    Libzip version => 1.0.1
